### PR TITLE
Subdivide primitive objects for performance optimization.

### PR DESCRIPTION
**Pull request by rongguodong**
Subdivided GVRCubeSceneObject. Each face is subdivided into NxN quads, where N is given by user.

GearVRf-DCO-1.0-Signed-off-by: Guodong Rong
rongguodong@hotmail.com

### CODE REVIEW

File: GVRf/Framework/src/org/gearvrf/scene_objects/GVRCubeSceneObject.java [hunk 1]
```diff
@@ -398,4 +436,415 @@ private void createSimpleCubeSixMeshes(GVRContext gvrContext,
...
+                subMeshes[index].setNormals(normals);
+                subMeshes[index].setTexCoords(texCoords);
+                subMeshes[index].setTriangles(indices);
+                grandchildren[index] = new GVRSceneObject(gvrContext,
+                        new FutureWrapper<GVRMesh>(subMeshes[index]),
+                        futureTextureList.get(5));
+                children[5].addChildObject(grandchildren[index]);
+            }
+        }
+
```
Comment (chenchao1407) [inline GVRf/Framework/src/org/gearvrf/scene_objects/GVRCubeSceneObject.java]: Still the same as before, the level of code duplicate is a little bit high. We may find some way to have a better code structure that can reuse the codes well.

**Comment by chenchao1407** [comment]
I think overall should be fine, may talk with you tomorrow to discuss about the code duplicate issue.

**Comment by rongguodong** [comment]
Code for subdivided sphere is checked in. Please review it.

**Comment by thomasflynn** [comment]
code looks good at first pass, just one change request:
'segmentNumber' sounds like you will be assigning a number to a segment.
I would prefer 'numberOfSegments' or 'numberOfSubdivisions' or something along those lines.

File: GVRf/Sample/gvr-cubemap/src/org/gearvrf/cubemap/CubemapScript.java [hunk 2]
```diff
@@ -62,6 +65,26 @@ public void onInit(GVRContext gvrContext) {
         GVRMaterial cubemapMaterial = new GVRMaterial(gvrContext,
                 GVRMaterial.GVRShaderType.Cubemap.ID);
         cubemapMaterial.setMainTexture(futureCubemapTexture);
+        
+        ArrayList<Future<GVRTexture>> futureTextureList = new ArrayList<Future<GVRTexture>>(6);
+        futureTextureList.add(gvrContext
+                .loadFutureTexture(new GVRAndroidResource(gvrContext,
+                        R.drawable.back)));
```
Comment (thomasflynn) [inline GVRf/Sample/gvr-cubemap/src/org/gearvrf/cubemap/CubemapScript.java]: you appear to be missing 'back', 'right', 'front', 'left', 'top', 'bottom' from your drawables directory.  they appear to not be checked-in.

**Comment by rongguodong** [comment]
Missing resources added.

**Comment by thomasflynn** [comment]
ok, turned on stats, and toggled frustum culling on/off.  appears to work as expected.

**Comment by rongguodong** [comment]
For variable names, I think both segmentNumber and numberOfSegment are quite clear in meaning. Since we also have sliceNumber, stackNumber, verticesNumber, etc. in other places, adding an extra "of" will only make all variable names longer.

**Comment by thomasflynn** [comment]
ok, anyone else have any other issues with this pull-request?

**Comment by chenchao1407** [comment]
I am currently busy with the following issue with #151. I think so far it is good to me now. May think of refactoring those duplicate codes later. But right now I think it is good to go.

**Pull request merged**